google/cloud/ndb/_cache.py

```diff
@@ -34,6 +34,14 @@
 log = logging.getLogger(__name__)
 
 
+def _syncpoint_692():
```

Comment:
I see the link in the testcase below to #692:  seems like it would be good to repeat here.  Also, I wonder about wrapping it in something like `if __debug__`, to avoid the overhead of even an empty python function call.

Reply:
e9e33fe

Reply:
5ec4bda

Poking around, it looks like you're correct about when `__debug__` is `True`. I'm not sure how common it is in practice to pass `-O` in production. I know I've never made a habit of it.

Another solution I thought about was getting rid of the no-op function and using something like `# pragma: SYNCPOINT syncname` to declare syncpoints, then do some `coverage` style postprocessing to insert the calls during testing. If you think the current implementation is mind-bending, though...

It's almost certainly not even something worth considering, unless `orchestrate` gets some kind of life outside of just `NDB`.

Reply:
Why not build this on `sys.settrace`?  Then you wouldn't have to modify the code under tests or cause extra calls and associated overhead.

You could pass in breakpoints for each functions as file-line locations to yield.

Reply:
Well, I didn't know about it, for starters. I can take a peek. On first blush, it seems like using line numbers is pretty fragile. I'd want a more robust way to define the syncpoint/breakpoint.

Reply:
Looks like I could probably use this to implement the `# pragma: SYNCPOINT` idea I mention elsewhere.

Reply:
I actually have something like this working now. Will try and make it presentable tomorrow.

Reply:
An interesting side-effect of using `sys.settrace` is it breaks `coverage`, since it also uses `sys.settrace`. ;-)

Reply:
19e207d